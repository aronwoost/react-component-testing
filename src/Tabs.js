import CustomButton from './CustomButton';
import React from 'react';

class Tabs extends React.Component {
  render() {
    const props = this.props;

    return (
      <div className="tabs">
        <CustomButton active={props.activeSection === "home"} label="Home" />
        <CustomButton active={props.activeSection === "about"} label="About" />
      </div>
    );
  }
}

export default Tabs;
