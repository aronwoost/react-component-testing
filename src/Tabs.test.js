import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Tabs from './Tabs';

it('renders without crashing', () => {
  const component = renderer.create(
    <Tabs activeSection='home' />
  );
  expect(component.toJSON()).toMatchSnapshot();
});
