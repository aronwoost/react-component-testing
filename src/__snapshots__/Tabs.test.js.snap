exports[`test renders without crashing 1`] = `
<div
  className="tabs">
  <button
    className="active">
    Home
  </button>
  <button
    className="">
    About
  </button>
</div>
`;
